docs/md/README.md

```diff
@@ -13,68 +13,166 @@ is a workload manager.
-other services as necessary to manage a workload.
+WFL is designed to be deployed to run as a service in the cloud, primarily
+on Kubernetes clusters.
 
-It can also be deployed to run as a service in the cloud.
+!!! info ""
+    For more on Workflow Launcher's role in the Terra infrastructure
+    see [Workflow Launcher's role in Terra](/docs/md/terra.md).
 
-For more on Workflow Launcher's role in the Terra infrastructure see
-[Workflow Launcher's role in Terra](/docs/md/terra.md).
 
-## Set up
+## Quickstart
+
+???+ tip
+    This is the Quickstart section, which should cover the most frequent
+    uses cases that interact with WFL. For more detailed information, please
+    check other sections such as the [development guide](/docs/md/dev.md)
+    or [modules design principles](/docs/md/module.md).
+
+### Build
+
+The easiest way to build WFL is via `make` (or `gmake` on macOS), in addition, the
+following prerequisites are needed:
+
+- [The Docker daemon](https://www.docker.com/products/docker-desktop)
+- Clojure (`brew install clojure` on macOS)
+- [Boot](https://github.com/boot-clj/boot) (`brew install boot-clj` on macOS)
+- Python3 (`brew install python@3.8` on macOS)
+- NodeJS (`brew install node` on macOS)
+
+!!! tip "Arch Linux tips"
+    - Install [clojure](https://www.archlinux.org/packages/?name=clojure) and
+    [leiningen](https://www.archlinux.org/packages/?name=leiningen)
+    from the official repositories.
+    - Install [boot](https://aur.archlinux.org/packages/boot/) and
+    [google-cloud-sdk](https://aur.archlinux.org/packages/google-cloud-sdk)
+    from the AUR.
+
+    You could then invoke `make` at the project level to test and build all
+    `workflow-launcher` modules:
 
-Invoke `make` at the project level to test and build all `workflow-launcher` modules:
 ```bash
-$ make -jN
+$ make -j8
 ```
-where `N` is the number of concurrent jobs you wish to run.
-
-`make` will build each module in `worfklow-launcher`, run tests and generate
+where `8` can be replaced by any number that represents the concurrent
+jobs you wish to run.
+
+!!! info
+    If the version of your `make` is above GNU Make 4.0 (you could check
+    by running `make --version`), it's **highly recommended** to use
+    `--output-sync` along with `-j` so the standard outputs are sorted, i.e.
+    ```
+    $ make -j8 --output-sync
+    ```
+
+`make` will **build** each module in `workflow-launcher`, **run tests** and **generate**
 `Docker` images. All generated files go into a `derived` directory under the
 project root.
 
-You can invoke `make` on a module from the top level directory by
+You can also invoke `make` on a module from the top level directory by
 
 ```bash
 $ make [MODULE] TARGET={prebuild|build|check|images|clean|distclean}
 ```
 
-For most of the time, you would want to run:
+where currently available `MODULE`s are {api cloud_function docs helm ui}
+
+For most of the time, you would want to run something like:
+
+```bash
+$ make distclean
```

Comment:
Note that distclean also removes 2p and 3p libraries too. I don't think this is always needed for common dev work. clean should be sufficient as it removes all derived module files. 

Perhaps create a jira to better document make targets and assign it to me.

Reply:
Also cleaning in parallel has nice speedups if you're running off an ssd or ramdisk